src/modules/commander/Commander.cpp

```diff
@@ -139,7 +139,7 @@ static float min_stick_change = 0.25f;
 static struct vehicle_status_s status = {};
 static struct actuator_armed_s armed = {};
 static struct safety_s safety = {};
-static int32_t _flight_mode_slots[manual_control_setpoint_s::MODE_SLOT_MAX];
+static int32_t _flight_mode_slots[manual_control_setpoint_s::MODE_SLOT_ARRAY_SIZE];
```

Comment:
This only needs to have `manual_control_setpoint_s::MODE_SLOT_ARRAY_SIZE -1`, right?

Reply:
No, array definition in C++ you give the number of elements that the array should have and since it's zero indexed it's one number higher than the highest index. Our highest index here is 6 and hence `..._ARRAY_SIZE` is 7 and the definition is correct.

Reference: http://www.cplusplus.com/doc/tutorial/arrays/

Reply:
But we only ever access 0..5 from what I can see.

Reply:
Right you are!

Reply:
ok, thanks for pointing that out because it resulted in the mode_slot overflow check not working properly. I fixed it by force-push in https://github.com/PX4/Firmware/compare/8a8d35b474e1fd9a0a8495726097a422457f7a17..1464492d43143c8ff4a020caebda35a997e852cf

The idea is to have the number of slots as a constant and the array for mapping the slots needs to have this size (as you correctly pointed out) and the slot number received in the commander can never be higher and it's also the number you use for the calculation in `rc_update`.